Выложил homework_10

AnMakar opened the pull request:
Макаров Антон Сергеевич, antonimak1997@gmail.com
На проверку @vauboy

Возникли серьезные проблемы с пониманием работы firewalld и iptables - вроде все настроил, в теории должно работать, но на практике возникают с этим серьезные проблемы. Но уж приходится выложить в том виде, который есть - сравнивать мне все равно не с чем.
К сожаленю из-за этого так и не смог добраться до extra

AnMakar committed:
Выложил homework_10

vauboy commented:
> Макаров Антон Сергеевич, [antonimak1997@gmail.com](mailto:antonimak1997@gmail.com) На проверку @vauboy
> 
> Возникли серьезные проблемы с пониманием работы firewalld и iptables - вроде все настроил, в теории должно работать, но на практике возникают с этим серьезные проблемы. Но уж приходится выложить в том виде, который есть - сравнивать мне все равно не с чем. К сожаленю из-за этого так и не смог добраться до extra

При добавлении сервиса ssh происходит добавление вот такого правила в iptables:
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:ssh ctstate NEW,UNTRACKED

Посмотрите в сторону rich rules.

vauboy commented on homework_10.sh:
```diff
@@ -0,0 +1,247 @@
+iptables -A INPUT -i enp0s8 -j REJECT
+iptables -A INPUT -i enp0s3 -j REJECT
+
+iptables -nvL
```
Добавьте вывод iptables -nvL

AnMakar replied:
```
Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination 
    0     0 ACCEPT     tcp  --  enp0s8 *       192.168.68.0/24      0.0.0.0/0            tcp dpt:22
    0     0 REJECT     all  --  enp0s8 *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable
   42  5996 REJECT     all  --  enp0s3 *       0.0.0.0/0            0.0.0.0/0            reject-with icmp-port-unreachable

Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination 

Chain OUTPUT (policy ACCEPT 50 packets, 6160 bytes)
 pkts bytes target     prot opt in     out     source               destination 

```

vauboy commented:
@AnMakar не посмотрели rich rules в firewalld?

AnMakar commented:
> @AnMakar не посмотрели rich rules в firewalld?

Как раз к этому времени добрался до домашнего ПК
Посмотрел как можно сделать с помощью rich rules. По идее должен работать такой вариант, если все правильно понял:

```
 firewall-cmd --permanent --zone=public --add-rich-rule 'rule family="ipv4" source address="192.168.68.0/24" service name="ssh" accept'
# success
firewall-cmd --permanent --zone=public --remove-service ssh
# success
firewall-cmd --permanent --zone=public --set-target=DROP
# success
firewall-cmd --reload
# success
firewall-cmd --list-all
# public (active)
#   target: DROP
#   icmp-block-inversion: no
#   interfaces: enp0s3
#   sources:
#   services: dhcpv6-client
#   ports:
#   protocols:
#   masquerade: no
#   forward-ports:
#   source-ports:
#   icmp-blocks:
#   rich rules:
#         rule family="ipv4" source address="192.168.68.0/24" service name="ssh" accept
firewall-cmd --zone=public --remove-interface=enp0s3
# success
firewall-cmd --zone=public --change-interface=enp0s8
# success
firewall-cmd --list-all
# public (active)
#   target: DROP
#   icmp-block-inversion: no
#   interfaces: enp0s8
#   sources:
#   services: dhcpv6-client
#   ports:
#   protocols:
#   masquerade: no
#   forward-ports:
#   source-ports:
#   icmp-blocks:
#   rich rules:
#         rule family="ipv4" source address="192.168.68.0/24" service name="ssh" accept
```

vauboy commented:
@AnMakar внесите изменения в домашку и можно мержить.

AnMakar committed:
Update homework_10.sh

Добавил вариант с сипользованием Rich Rules и добавил вывод команды iptables -nvL
